<launch>
    <arg name="map_file_name0" default="none"/>
    <arg name="map_file_name1" default="none"/>
    <arg name="map_file_name2" default="none"/>
    <node pkg="tf2_ros" exec="static_transform_publisher" args="--x 0 --y 0 --z 0 --yaw 0 --pitch 0 --roll 0 --frame-id world --child-frame-id map"/>
    <node pkg="rviz2" exec="rviz2" args="-d $(find-pkg-share ros2_orb_slam3)/config/compare_point_clouds.rviz"/>
    <node pkg="ros2_orb_slam3" exec="compare_point_clouds">
        <param name="map_file_name0" value="$(var map_file_name0)"/>
        <param name="map_file_name1" value="$(var map_file_name1)"/>
        <param name="map_file_name2" value="$(var map_file_name2)"/>
    </node>
	<node pkg="octomap_server" exec="octomap_server_node" name="octomap_server">
		<param name="resolution" value="0.01" />

		<!-- fixed map frame (set to 'map' if SLAM or localization running!) -->
		<param name="frame_id" value="map" />

		<!-- maximum range to integrate (speedup!) -->
		<param name="sensor_model.max_range" value="5.0" />

        <param name="filter_ground" value="true" />

		<!-- data source to integrate (PointCloud2) -->
		<remap from="cloud_in" to="orb_point_cloud2" />
	</node>
</launch>
